support-lib/jni/djinni_support.cpp

```diff
@@ -28,12 +28,35 @@ namespace djinni {
 // Set only once from JNI_OnLoad before any other JNI calls, so no lock needed.
 static JavaVM * g_cachedJVM;
 
+/*static*/
+JniClassInitializer::registration_vec & JniClassInitializer::get_vec() {
+    static JniClassInitializer::registration_vec m;
+    return m;
+}
+
+/*static*/
+std::mutex & JniClassInitializer::get_mutex() {
+    static std::mutex mtx;
+    return mtx;
+}
+
+/*static*/
+JniClassInitializer::registration_vec JniClassInitializer::get_all() {
+    const std::lock_guard<std::mutex> lock(get_mutex());
+    return get_vec();
+}
+
+JniClassInitializer::JniClassInitializer(const std::function<void()> & init) {
+    const std::lock_guard<std::mutex> lock(get_mutex());
+    get_vec().emplace_back(init);
```

Comment:
One more thing noticed on final review.  This function should take its arg by value, and then use std::move here.  Since in the normal use case the input is a temporary std::function (created from a function pointer) this will avoid an extra copy (which involves heap allocation in the case of std::function).